opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1644,6 +1698,13 @@ namespace Opm {
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] /= refDens;
                     cellRes[ flowPhaseToEbosCompIdx( flowPhaseIdx ) ] *= cellVolume;
                 }
+                if (has_solvent_) {
+                    // no need to store refDens for all cells?
+                    const auto& intQuants = ebosSimulator_.model().cachedIntensiveQuantities(cellIdx, /*timeIdx=*/0);
```

Comment:
yes, we should really consider relying on the IQ cache to be unsafe. the alternative is to iterate over the grid and use element contexts to access the IQs like e.g. [here](https://github.com/OPM/opm-simulators/blob/master/opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp#L673-L688). That said, this PR does not change anything for the worse on that front...